usageReports/rds/prepare_response.go

```diff
@@ -0,0 +1,159 @@
+						} `json:"hits"`
+					} `json:"hits"`
+				} `json:"top_reports_hits"`
+			} `json:"buckets"`
+		} `json:"top_reports"`
+	}
+)
+
+// addCostToReport adds cost for each instance based on billing data
+func addCostToReport(report rds.Report, costs ResponseCost) (rds.Report) {
+	for _, accounts := range costs.Accounts.Buckets {
+		if accounts.Key != report.Account {
+			continue
+		}
+		for _, instance := range accounts.Instances.Buckets {
+			for i := range report.Instances {
+				if report.Instances[i].CostDetail == nil {
+					report.Instances[i].CostDetail = make(map[string]float64, 0)
+				}
+				split := strings.Split(instance.Key, ":")
+				if len(split) == 7 && split[6] == report.Instances[i].DBInstanceIdentifier {
+					report.Instances[i].Cost += instance.Cost.Value
+					report.Instances[i].CostDetail["instance"] += instance.Cost.Value
+				}
+			}
+		}
+	}
+	return report
+}
+
+// prepareResponseRdsDaily parses the results from elasticsearch and returns the RDS report
+func prepareResponseRdsDaily(ctx context.Context, resRds *elastic.SearchResult, resCost *elastic.SearchResult) ([]rds.Report, error) {
+	parsedRds  := ResponseRds{}
+	cost := ResponseCost{}
+	reports := []rds.Report{}
+	err := json.Unmarshal(*resRds.Aggregations["top_reports"], &parsedRds.TopReports)
+	if err != nil {
+		return nil, err
+	}
+	if resCost != nil {
+		json.Unmarshal(*resCost.Aggregations["accounts"], &cost.Accounts)
+	}
+	for _, account := range parsedRds.TopReports.Buckets {
+		if len(account.TopReportsHits.Hits.Hits) > 0 {
+			report := account.TopReportsHits.Hits.Hits[0].Source
+			report = addCostToReport(report, cost)
+			reports = append(reports, report)
+		}
+	}
+	return reports, nil
+}
+
+// prepareResponseRdsMonthly parses the results from elasticsearch and returns the RDS usage report
+func prepareResponseRdsMonthly(ctx context.Context, resRds *elastic.SearchResult) ([]rds.Report, error) {
+	response := ResponseRds{}
+	reports := []rds.Report{}
+	err := json.Unmarshal(*resRds.Aggregations["top_reports"], &response.TopReports)
+	if err != nil {
+		return nil, err
+	}
+	for _, account := range response.TopReports.Buckets {
+		if len(account.TopReportsHits.Hits.Hits) > 0 {
+			reports = append(reports, account.TopReportsHits.Hits.Hits[0].Source)
+		}
+	}
+	return reports, nil
+}
+
+func isInstanceUnused(instance rds.Instance) bool {
+	average := instance.CpuAverage
+	peak    := instance.CpuPeak
+	if peak >= 60.0 {
+		return false
+	} else if average >= 10.0 {
+		return false
+	}
+	return true
+}
+
+// prepareResponseRdsUnused filter reports to get the top instances
+func prepareResponseRdsUnused(params rdsUnusedQueryParams, reports []rds.Report) (int, []rds.Instance, error) {
+	instances := []rds.Instance{}
+	for _, report := range reports {
+		for _, instance := range report.Instances {
+			if isInstanceUnused(instance) {
+				instances = append(instances, instance)
+			}
+		}
+	}
+	for i := 0; i < len(instances) - 1; i++ {
```

Comment:
There are function in Golang that allows you to sort slices (https://golang.org/pkg/sort/#Slice). Please use it instead of making a sort algorithm here